test/real_content/swupd_update.bats

```diff
@@ -1,156 +1,13 @@
-	done
-}
-
-test_setup() {
-	# Fill defaults in optional parameters
-	if [ -z "$URL" ]; then
-		URL=https://cdn.download.clearlinux.org/update
-	fi
-	if [ -z "$MAX_VERSIONS" ]; then
-		MAX_VERSIONS=10
-	fi
-
-	# Using github clr-bundles project to get official clear releases
-	# version list is sorted from newer to older
-	# TODO: Discover available versions to work with custom mixers
-	read -r -d '\n' -a version_list < <(git ls-remote --tags https://github.com/clearlinux/clr-bundles.git | grep -v {} | grep -v latest | sed "s/.*refs\\/tags\\///" | sort -gr | head -n "$MAX_VERSIONS" | sort -g) || echo
-	if [ "${#version_list[@]}" -eq 0 ]; then
-		echo "Impossible to get versions list"
-		return 1
-	fi
-
-	# Get the last format
-	FORMAT=$(get_format_from_versions "${version_list[@]}")
-	export FORMAT
-
-	# Fill a list of last available versions (up to ${MAX_VERSIONS})
-	i=0
-	for v in "${version_list[@]}"; do
-		format=$(get_format "$v")
-		# if version isn't published or is from a different version, discard
-		if [ -z "$format" ] || [ "$FORMAT" -ne "$format" ]; then
-			continue
-		fi
-
-		VERSION[$i]=$v
-		i=$((i+1))
-	done
-
-	# Not enough versions to run the test
-	if [ -z "${VERSION[0]}" ] || [ -z "${VERSION[1]}" ]; then
-		# TODO: This is going to break on first release in a new format
-		echo "We need at least 2 versions in format $FORMAT to continue with this test"
-		return 1
-	fi
-
-	sudo rm -rf "$TEST_NAME"
-	sudo mkdir -p "$TEST_NAME"
-	ROOT_DIR="${PWD}/${TEST_NAME}"
-
-	# TODO: use -S ${ROOT_DIR}/var/lib/swupd/ when #665 gets fixed
-	# TODO: Add -t
-	SWUPD_OPTS_SHORT="-u ${URL} -p ${ROOT_DIR} -S ${ROOT_DIR}/swupd/"
-	SWUPD_OPTS="$SWUPD_OPTS_SHORT --no-scripts"
-}
-
-test_teardown() {
-	sudo rm -rf "$TEST_NAME"
-}
-
-verify_system() {
-	# TODO: This is a bug in verify --picky #632
-	sudo rm -f "${ROOT_DIR}/usr/share/clear/bundles/.MoM"
-
-	run sudo sh -c "$SWUPD verify --picky $SWUPD_OPTS 2>/dev/null"
-	if [ -n "$output" ]; then
-		echo "Verify found extra files in the system:"
-		echo "$output"
-		return 1
-	fi
-
-	run sudo sh -c "$SWUPD verify $SWUPD_OPTS"
-	assert_not_in_output "did not match"
-
-	return
-}
-
-# check_version: Check if version in /usr/lib/os-release is equal $1.
-# Fails otherwise.
-check_version() {
-	version="$1"
-	cur_version=$(grep VERSION_ID "${ROOT_DIR}/usr/lib/os-release" | cut -d = -f 2)
-	if [ "$version" -ne "$cur_version" ]; then
-		echo "Version $version is different from expected $cur_version"
-		return 1
-	fi
-}
-
-install_bundles() {
-	if [ -z "$BUNDLE_LIST" ]; then
-		run sudo sh -c "$SWUPD bundle-list --all $SWUPD_OPTS_SHORT"
-		BUNDLE_LIST=$(echo "$output" | tr '\n' ' ')
-	fi
-
-	echo "Install bundles: $BUNDLE_LIST"
-
-	run sudo sh -c "$SWUPD bundle-add $SWUPD_OPTS --skip-diskspace-check $BUNDLE_LIST"
-	assert_status_is 0
-	verify_system
-}
-
-@test "test bundle-list --all" {
+# Test if bundle-list --all is working because all other tests depend on that
+@test "RC001: test bundle-list --all" {
```

Comment:
This file is called swupd_update, but this is a bundle-list test. Does it make sense to move this one?

Reply:
The only reason I'm testing a bundle-list here is that if bundle-list is failing we can't trust the results of the tests. And I'd like to run that before next test.

Reply:
It is not a good practice to have tests that depend from other tests, but sometimes it can be necessary. So if you think the result of tests RC002-04 depend on the good execution of RC001 maybe you can create a file like .RC001 at the end of the first test, and run the following tests only if .RC001 exists. Then you can remove .RC001 in the global_teardown (and in the global_setup in case it was left over for some reason).

Reply:
Ok, maybe I wasn't clear in my comments about that test. Actually this test isn't really necessary and it looks out of scope for real content tests. We could remove it, but there's a reason I added that.

Almost all real content tests count on bundle-list --all to get a list of bundles to be tested. If by any change this command is broken and it's returning, for example, just one bundle. In this case we would run the tests only on that bundle and even if all tests are successful we can't assure that swupd is working fine for other cases.

So we don't need to run RC001 before any other test, so I didn't add it as a dependency (like we do with generate-cert). No sure if blocking other tests is the best way to go on this case.

Do you think we really need to have a strategy like you suggested.

Reply:
I realized we don't need to have this test here. We can test that every time because it's not time consuming, so we can get better results. I'll remove RC001 and improve the other tests.